src/components/TokenAdministrative.js

```diff
@@ -178,7 +190,7 @@ class TokenAdministrative extends React.Component {
         case 'mint':
           return <TokenMint htrBalance={this.props.htrBalance} action={this.state.action} cancelAction={this.cancelAction} token={this.props.token} showSuccess={this.showSuccess} />
         case 'melt':
-          return <TokenMelt action={this.state.action} cancelAction={this.cancelAction} token={this.props.token} showSuccess={this.showSuccess} walletAmount={this.state.balance} />
+          return <TokenMelt action={this.state.action} cancelAction={this.cancelAction} token={this.props.token} showSuccess={this.showSuccess} walletAmount={this.state.balance.data.available} />
```

Comment:
Actually, we would never see the loading because the screen that displays this component (`src/screens/Wallet.js`) will only be rendered after the balance and history for the `selectedToken` gets downloaded

Reply:
Added a loading on the `TokenMelt` component if we ever decide to use it elsewhere

29502d683bebecad85c540c7caad1c814f2066f2